setup.py

```diff
@@ -27,7 +29,70 @@
     )
 
 readme_md_path = topdir_from_setup / "README.md"
-long_description = readme_md_path.read_text("utf-8")
+readme_text = readme_md_path.read_text("utf-8")
+
+
+########################################################################
+# Replace repo relative URLs in README.md with absolute URLs pointing
+# to github.com. To test this locally, try something like the
+# following:
+#
+#   $ python3 setup.py --long-description > README-filtered.md
+#   $ diff -u README.md README-filtered.md | cdiff
+#   $ pandoc -s --metadata title=README README-filtered.md -o README-filtered.html
+#   $ firefox README-filtered.html
+#
+########################################################################
+
+git_ref = f"v{const.VERSION}"
+git_ref = "v0.4.92a2"
+git_ref = "main"
```

Comment:
This is the hardest part of this idea - What git_ref should we use to ground the links?

I think we should figure out a good release workflow, and the answer will follow.  I like the flow of:
- Bump const.VERSION to whatever we want the release to be, and commit it.
- `git tag -s v...` to match and push the tag
- In Github, make a "Release" based on the tag  (maybe this can be combined so github makes the tag?  Can it sign the tag?)
- A GitHub action does the build and pushes it to pypi for us (this works; I published the v0.4.92a2 package to pypi using this mechanism already)

The only caveat is that the version that gets published to pypi doesn't match the git tag; only the `const.VERSION` in case there's a difference. I don't like that there's that drift...  Maybe we can work on tooling to make sure they're in sync for releases...

If we go with something like that, we can use `f"v{const.VERSION}"` because we'll always be guaranteeing that it always matches with a tag on github.

Reply:
So the data going into the long_description metadata would come from:

  * The `README.md` from the filesystem.
  * The current `git_ref`, which may contain the current `const.VERSION`.

The second item rules out generating `long_description.md` and committing it to the repo, because it requires us to re-run the conversion every time any of the data has changed and someone needs to know the current `long_description` - which means when `setup.py` runs.

If the `setup.py` script runs in the context of a git repo though, the `setup.py` script can determine that and find out the current git ref, and use that to generate proper links. That git ref could be a symbolic ref like a tag or a branch name, but it could also be the commit sha. Especially using the sha for the current commit would make sure that the links point to the exact same file versions current at the time of the `setup.py` run.

Of course, then the question remains of what to do when `setup.py` is running from e.g. an extracted dist tarball instead of a git clone. I would say to just link to the `main` branch in that case.